R/flatten.R

```diff
@@ -58,11 +58,17 @@ flatten <- function(tbl, groups, wide_cols, sep = "_") {
   rename <- colnames(out[[1]]) %in% wide_cols
   for (grp in names(out)) {
     group <- out[[grp]]
-    colnames(group)[rename] <- paste0(grp, sep, colnames(group)[rename])
+    colnames(group)[rename] <- if (any("x" == colnames(group))) {
+      gsub("x(_)(.*)|^x$", paste0("\\2", "\\1", extend_normalization_type_name(grp)),
+           colnames(group)[rename])
+      } else {
+        paste0(grp, sep, colnames(group)[rename])
+      }
     out[[grp]] <- group
   }
 
   ## Drop empty elements for successful merge.
   filtered <- out[lapply(out, nrow) > 0L]
-  Reduce(function(x, y) x[y, on = intersect(names(x), names(y))], filtered)
+  Reduce(function(x, y) x[y, on = intersect(names(x), names(y)),
+                          allow.cartesian = TRUE], filtered)
```

Comment:
1. lintr said or 
```
Reduce(function(x, y) x[y, on = intersect(names(x), names(y)), allow.cartesian = TRUE], filtered)
```
or use  `function(x,y) {...}`
2. Are we sure we will allow cartesian?